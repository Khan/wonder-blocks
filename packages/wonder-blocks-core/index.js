// @flow
import type {
    ClickableHandlers,
    ClickableState,
} from "./components/clickable-behavior.js";
import type {AriaProps, IIdentifierFactory, StyleType} from "./util/types.js";
import type {Intersection} from "./util/get-element-intersection.js";

export {default as ClickableBehavior} from "./components/clickable-behavior.js";
export {default as Text} from "./components/text.js";
export {default as View} from "./components/view.js";
export {
    default as WithSSRPlaceholder,
} from "./components/with-ssr-placeholder.js";
export {default as IDProvider} from "./components/id-provider.js";
export {default as UniqueIDProvider} from "./components/unique-id-provider.js";
export {default as addStyle} from "./util/add-style.js";
export {
    default as getClickableBehavior,
} from "./util/get-clickable-behavior.js";
export {
    default as getElementIntersection,
} from "./util/get-element-intersection.js";

export type {
    AriaProps,
    ClickableHandlers,
    ClickableState,
    Intersection,
    IIdentifierFactory,
    StyleType,
};
