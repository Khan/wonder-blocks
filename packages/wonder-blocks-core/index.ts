// @flow
import {ClickableHandlers} from "./components/clickable-behavior";
import {AriaProps, IIdentifierFactory, StyleType} from "./util/types";
import {Intersection} from "./util/get-element-intersection";

export {default as ClickableBehavior} from "./components/clickable-behavior";
export {default as Text} from "./components/text";
export {default as View} from "./components/view";
export {
    default as WithSSRPlaceholder,
} from "./components/with-ssr-placeholder";
export {default as UniqueIDProvider} from "./components/unique-id-provider";
export {default as addStyle} from "./util/add-style";
export {
    default as getClickableBehavior,
} from "./util/get-clickable-behavior";
export {
    default as getElementIntersection,
} from "./util/get-element-intersection";

export {
    AriaProps,
    ClickableHandlers,
    Intersection,
    IIdentifierFactory,
    StyleType,
};
