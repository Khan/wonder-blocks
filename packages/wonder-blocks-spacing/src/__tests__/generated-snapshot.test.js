// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests:
//   1. edit the markdown documentation files in the package,
//        packages/wonder-blocks-spacing
//   2. Run `yarn run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";

// Mock react-dom as jest doesn't like findDOMNode.
jest.mock("react-dom");
import {StyleSheet} from "aphrodite";
import {View} from "@khanacademy/wonder-blocks-core";
import Spacing from "../../src/index.js";

describe("wonder-blocks-spacing", () => {
    it("example 1", () => {
        const styles = StyleSheet.create({
            row: {
                flexDirection: "row",
                alignItems: "center",
                marginBottom: Spacing.xxSmall_6,
            },
        });
        const example = (
            <View>
                {Object.keys(Spacing).map((spaceName, idx) => (
                    <View key={idx} style={styles.row}>
                        <View
                            style={{
                                width: 250,
                                textAlign: "right",
                            }}
                        >
                            {spaceName}: {Spacing[spaceName]}px
                        </View>
                        <View
                            style={{
                                width: Spacing.xSmall_8,
                            }}
                        />
                        <View
                            style={{
                                width: Spacing.xxxLarge_64,
                            }}
                        >
                            <View
                                style={{
                                    backgroundColor: "black",
                                    width: Spacing[spaceName],
                                    height: Spacing.xxxSmall_4,
                                }}
                            />
                        </View>
                        <View
                            style={{
                                width: Spacing.xSmall_8,
                            }}
                        />
                        <View
                            style={{
                                backgroundColor: "black",
                                width: Spacing.xxxSmall_4,
                                height: Spacing[spaceName],
                            }}
                        />
                    </View>
                ))}
            </View>
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
