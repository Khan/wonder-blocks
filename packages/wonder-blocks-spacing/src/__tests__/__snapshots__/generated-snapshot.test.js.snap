// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-spacing example 1 1`] = `
<div
  className=""
  style={
    {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xxxxSmall_2
      : 
      2
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 2,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 2,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xxxSmall_4
      : 
      4
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 4,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 4,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xxSmall_6
      : 
      6
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 6,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 6,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xSmall_8
      : 
      8
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 8,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 8,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      small_12
      : 
      12
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 12,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 12,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      medium_16
      : 
      16
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 16,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 16,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      large_24
      : 
      24
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 24,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 24,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xLarge_32
      : 
      32
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 32,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 32,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xxLarge_48
      : 
      48
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 48,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 48,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
  <div
    className=""
    style={
      {
        "alignItems": "center",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "row",
        "margin": 0,
        "marginBottom": 6,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "textAlign": "right",
          "width": 250,
          "zIndex": 0,
        }
      }
    >
      xxxLarge_64
      : 
      64
      px
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 64,
          "zIndex": 0,
        }
      }
    >
      <div
        className=""
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "boxSizing": "border-box",
            "display": "flex",
            "flexDirection": "column",
            "height": 4,
            "margin": 0,
            "minHeight": 0,
            "minWidth": 0,
            "padding": 0,
            "position": "relative",
            "width": 64,
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 8,
          "zIndex": 0,
        }
      }
    />
    <div
      className=""
      style={
        {
          "alignItems": "stretch",
          "backgroundColor": "black",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "height": 64,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "width": 4,
          "zIndex": 0,
        }
      }
    />
  </div>
</div>
`;
